app/controllers/TaskController.scala

```diff
@@ -368,6 +370,21 @@ class TaskController @Inject() (implicit val env: Environment[User, SessionAuthe
             GSVLinkTable.save(gsvLink)
           }
         }
+
+        val visitedTimestamp: Long = pano.visitedTimestamp.map(value => value).getOrElse(0)
```

Comment:
This is the time to convert from a `Long` to a `Timestamp`.

And if the value can be null (not sure if it should, but if that is something we need to account for), then you want an `Option[Timestamp]` here. You've taken something with more clear information (`Option[Long]`) and turned it into something that is less clear (a `Long` where 0 means null).